doc/high_availability.md

```diff
@@ -1,110 +1,74 @@
-are not enough voting machines.
- 
- The EnableHA API call will report will report the changes that it 
-made to the model, which will shortly be reflected in reality
-### The API 
-
- There is an API server running on all State Machines, these talk to all
-the peers but queries and updates are addressed to the mongo master instance.
- 
- Unit and machine agents connect to any of the API servers, by trying to connect
-to all the addresses concurrently, but not simultaneously. It starts to try each
-address in turn after a short delay. After a successful connection, the
-connected address will be stored; it will be tried first when next connecting.
-
-### The peergrouper worker:
- 
- It looks at the current state and decides what the peergroup members should 
-look like and continually tries to maintain those members.
-
- The reason for its existence is that it can often take a while for mongo to 
-allow a peer group change, so we can't change it directly in the 
-EnableHA API call
-
- Its worker loop continally watches 
-
- 1. The current set of controllers 
- 2. The addresses of the current controllers 
- 3. The status of the current mongo peergroup
- 
-It feeds all that information into `desiredPeerGroup`, which provides the peer 
-group that we want to be and continually tries to set that peer group in mongo 
-until it succeeds.
- 
-**NOTE:** There is one situation which currently doesn't work which is 
-that if you've only got one controller, you can't switch to another one.
-
-### The Singleton Workers
-
-**Note:** This section reflects the current behavior of these workers but 
-should by no means be taken as an example to follow since most (if not all)
-should run concurrently and are going to change in the near future.
-
-The following workers require only a single instance to be running
-at any one moment:
-
- * The environment provisioner
- * The firewaller
- * The charm revision updater
- * The state cleaner
- * The transaction resumer
- * The minunits worker
-
-When a machine agent connects to the state, it decides whether
-it is on the same instance as the mongo master instance, and
-if so, it runs the singleton workers; otherwise it doesn't run them.
-
-Because we are using `mgo.Strong` consistency semantics,
-it's guaranteed that our mongo connection will be dropped
-when the master changes, which means that when the
-master changes, the machine agent will reconnect to the
-state and choose whether to run the singleton workers again.
-
-It also means that we can never accidentally have two
-singleton workers performing operations at the same time.
+HA mode is invoked via the `enable-ha` command. By default, this will ensure 
+three controllers, but using the `-n` flag will allow running with five or 
+seven controllers.
+
+The number of controllers can be reduced by invoking `remove-machine` with a 
+controller machine ID, and increased by re-running the `enable-ha` command.
```

Comment:
I certainly think that long term this will be nice to become `juju add-unit -m controller juju-controller` and/or `juju remove-unit`.
Someday :)

Reply:
For 4.0, I really was hoping we'd do this TBH, especially now the controller charm does what it's supposed to do wrt configuring dqlite etc. Is there a concrete reason why we can't now drop enable-ha in favour of add/remove unit?

Reply:
Other than hoping to get it done one day, no.